### PR TITLE
add "type" property in projections by default to hide permission warnings

### PR DESCRIPTION
**Pull request by ETLaurent**
## Summary

Improve the `can` method of the permission module in order to hide warnings by always providing to it the `type` property of a doc or widget and getting the appropriate manager.

## What are the specific steps to test this change?

Log in as an editor.
Create a new page with a slug `/a`.
Create another page and type `/a` in the slug field
There should be no `type has no manager` error in the server console

## What kind of change does this PR introduce?
*(Check at least one)*

- [x] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [x] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/permission/index.js [hunk 1]
```diff
@@ -61,7 +61,11 @@ module.exports = {
         }
         const type = docOrType && (docOrType.type || docOrType);
         const doc = (docOrType && docOrType._id) ? docOrType : null;
-        const manager = type && self.apos.doc.getManager(type);
+        let manager = type && self.apos.doc.getManager(type);
+        if (doc && !manager) {
+          // Fallback to an appropriate manager module
+          manager = self.apos.util.getManagerOf(doc);
```
Comment (boutell) [inline modules/@apostrophecms/permission/index.js:67]: We don't support viewRole etc. for widget types. This shouldn't be necessary. We should not be calling permission.can with a widget ever actually. Are we doing that though?

**Review by boutell — CHANGES_REQUESTED** [review]
I think maybe we got a little past the requirements here and did something that doesn't make sense, but you may have seen something I haven't seen, so let me know.

File: modules/@apostrophecms/permission/index.js (hunk 1 above)
Comment (ETLaurent) [reply modules/@apostrophecms/permission/index.js:67]: yes, in the annotate function:

```js
      annotate(req, action, objects) {
        const property = `_${action}`;
        for (const object of objects) {
          if (self.can(req, action, object)) {
            object[property] = true;
          }
        }
      },
```

we could do something like that to exclude widgets, WDYT?

```js
      annotate(req, action, objects) {
        const property = `_${action}`;
        for (const object of objects) {
          if (object.metaType !== 'widget' && self.can(req, action, object)) {
            object[property] = true;
          }
        }
      },
```

Comment (boutell) [reply modules/@apostrophecms/permission/index.js:67]: I think we would have to check that the `metaType` is `doc`, to exclude other things that won't work, but yes I agree with this approach.

**Review by boutell — CHANGES_REQUESTED** [review]
I think we would have to check that the metaType is doc, to exclude other things that won't work, but yes I agree with this approach.

**Review by haroun — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**